### PR TITLE
Added Header modification check

### LINKED ISSUE #40: Project doesn't build despite header files changing
For example, let's say we have ```Test.h``` and ```Test.cpp```.
We build the project, then change ```Test.h```.
We try to build the project again, only to be told the project need not be rebuilt.

This is a bug, as changes in the header file can have a big impact on the actual code (post pre-processing).

### PR DESCRIPTION
**Pull request by aTOz132**
This closes #40.

### CODE REVIEW

File: scripts/build.py [hunk 1]
```diff
@@ -106,14 +106,21 @@ def should_compile(file):
     # Check if a matching object file exists and that the original file is compile-valid.
     if path.splitext(file)[-1] in COMPILE_VALID_EXT:
         if path.exists(path.join('obj', path.splitext(file)[0] + OBJECT_EXT)):
-            # Compare modification times
-            return stat(file).st_mtime > stat(path.join('obj', path.splitext(file)[:-1][0] + OBJECT_EXT)).st_mtime
+            # Compare modification times of .cpp vs obj OR .h vs obj has been changed
+            if stat(file).st_mtime > stat(
```
Comment (ExhoAR22) [inline scripts/build.py]: Don't leave function calls open like that.
Take the entire call one line down if you have to.

File: scripts/build.py [hunk 2]
```diff
@@ -106,14 +106,21 @@ def should_compile(file):
...
+             path.join('obj', path.splitext(file)[:-1][0] + OBJECT_EXT)).st_mtime:
+                return True
+            elif path.exists(path.splitext(file)[0]+".h") and stat(
```
Comment (ExhoAR22) [inline scripts/build.py]: Your spacing is inconsistent.
```...[0]+".h"``` -> ```...[0] + '.h'```

Comment (ExhoAR22) [inline scripts/build.py]: Also, don't leave function calls open like you did with ```stat``` here.

File: scripts/build.py [hunk 3]
```diff
@@ -106,14 +106,21 @@ def should_compile(file):
...
+             (path.splitext(file)[0])+".h").st_mtime > stat(path.join('obj', path.splitext(
```
Comment (ExhoAR22) [inline scripts/build.py]: I believe these lines should be slightly more indented, but I may be mistaken.
What do PyCharm and pylint say?

File: scripts/build.py (hunk 1 above)
Comment (aTOz132) [reply scripts/build.py]: vscode only allows me to newline it like this with this parenthesis game but ill try

File: scripts/build.py (hunk 3 above)
Comment (aTOz132) [reply scripts/build.py]: check the new commit, i fixed according to pep8

File: scripts/build.py (hunk 2 above)
Comment (aTOz132) [reply scripts/build.py]: fixed

Comment (aTOz132) [reply scripts/build.py]: fixed

File: scripts/build.py (hunk 1 above)
Comment (aTOz132) [reply scripts/build.py]: fixed.

File: scripts/build.py [hunk 4]
```diff
@@ -106,14 +106,21 @@ def should_compile(file):
     # Check if a matching object file exists and that the original file is compile-valid.
     if path.splitext(file)[-1] in COMPILE_VALID_EXT:
         if path.exists(path.join('obj', path.splitext(file)[0] + OBJECT_EXT)):
-            # Compare modification times
-            return stat(file).st_mtime > stat(path.join('obj', path.splitext(file)[:-1][0] + OBJECT_EXT)).st_mtime
+            # Compare modification times of .cpp vs obj OR .h vs obj has been changed
+            if stat(file).st_mtime
+            > stat(path.join('obj', path.splitext(file)[:-1][0] + OBJECT_EXT)).st_mtime:
```
Comment (ExhoAR22) [inline scripts/build.py]: Missing indentation

File: scripts/build.py [hunk 5]
```diff
@@ -106,14 +106,21 @@ def should_compile(file):
...
+                return True
+            elif path.exists(path.splitext(file)[0]+'.h') and stat((path.splitext(file)[0])+".h").st_mtime
+            > stat(path.join('obj', path.splitext(
```
Comment (ExhoAR22) [inline scripts/build.py]: Missing indentation

File: scripts/build.py (hunk 4 above)
Comment (aTOz132) [reply scripts/build.py]: fixed

File: scripts/build.py (hunk 5 above)
Comment (aTOz132) [reply scripts/build.py]: fixed

**Pull request merged**